### PR TITLE
[FIX] base: translations lost after synchronization

### PR DESCRIPTION
**Pull request by rco-odoo**
Modifying a source term in an XML/HTML translated field can lose translations
if the same term is translated in several languages.

### CODE REVIEW

**Comment by rco-odoo** [comment]
I made explicit pull requests for other branches, in order to handle special cases in those versions:
- #29081 (translation type is different in 12.0)
- #29082 (missing translations have no value in master)

File: odoo/addons/base/ir/ir_translation.py [hunk 1]
```diff
@@ -464,41 +464,48 @@ def _sync_terms_translations(self, field, records):
...
+        discarded = Translation
 
         for record in records:
             # get field value and terms to translate
             value = record[field.name]
             terms = set(field.get_trans_terms(value))
-            record_trans = trans.search([
-                ('type', '=', 'model'),
+            translations = Translation.search([
+                ('type', '=', 'model_terms'),
```
Comment (d-fence) [inline odoo/addons/base/ir/ir_translation.py]: I don't see 'model_terms' in TRANSLATION_TYPE ?

Comment (KangOl) [inline odoo/addons/base/ir/ir_translation.py]: Not for `10.0`

**Comment by rco-odoo** [comment]
robodoo r+

**Pull request closed without merging**